From 7bc6842d4c6de588919de9fd42ddb54af00fcb62 Mon Sep 17 00:00:00 2001
From: Dan Prince <dprince@redhat.com>
Date: Wed, 13 Mar 2024 07:50:05 -0400
Subject: [PATCH 2/6] FIPS-compliant Dockerfile updates

This commit modifies the Dockerfile to have the rabbitmq-cluster-operator
to be built from ubi9 go-toolset and ubi-minimal images so to have
the necessary dependencies to be FIPS compliant.

It also squashes the build steps by removing the intermediate step
where we create the respective user/group.
---
 Dockerfile | 23 +++++++++--------------
 1 file changed, 9 insertions(+), 14 deletions(-)

diff --git a/Dockerfile b/Dockerfile
index eb648a8..2dcbcbe 100644
--- a/Dockerfile
+++ b/Dockerfile
@@ -1,8 +1,11 @@
+
 # Build the manager binary
-FROM --platform=$BUILDPLATFORM golang:1.20 as builder
+FROM --platform=$BUILDPLATFORM registry.access.redhat.com/ubi9/go-toolset:1.20 as builder
 
 WORKDIR /workspace
 
+USER root
+
 # Dependencies are cached unless we change go.mod or go.sum
 COPY go.mod go.mod
 COPY go.sum go.sum
@@ -20,26 +23,18 @@ ARG TARGETOS
 ARG TARGETARCH
 ENV GOOS $TARGETOS
 ENV GOARCH $TARGETARCH
-RUN CGO_ENABLED=0 GO111MODULE=on go build -a -tags timetzdata -o manager main.go
-
-# ---------------------------------------
-FROM alpine:latest as etc-builder
+RUN CGO_ENABLED=1 GO111MODULE=on go build -a -tags 'strictfipsruntime timetzdata' -o manager main.go
 
-RUN echo "rabbitmq-cluster-operator:x:1000:" > /etc/group && \
-    echo "rabbitmq-cluster-operator:x:1000:1000::/home/rabbitmq-cluster-operator:/usr/sbin/nologin" > /etc/passwd
-
-RUN apk add -U --no-cache ca-certificates
-
-# ---------------------------------------
-FROM scratch
+# ---
+FROM registry.access.redhat.com/ubi9/ubi-minimal:latest
 
 ARG GIT_COMMIT
 LABEL GitCommit=$GIT_COMMIT
 
 WORKDIR /
 COPY --from=builder /workspace/manager .
-COPY --from=etc-builder /etc/passwd /etc/group /etc/
-COPY --from=etc-builder /etc/ssl/certs/ca-certificates.crt /etc/ssl/certs/ca-certificates.crt
+RUN echo "rabbitmq-cluster-operator:x:1000:" > /etc/group && \
+    echo "rabbitmq-cluster-operator:x:1000:1000::/home/rabbitmq-cluster-operator:/usr/sbin/nologin" > /etc/passwd
 
 USER 1000:1000
 
-- 
2.45.2

